src/cursor.c

```diff
@@ -160,6 +165,10 @@ static void CursorList_IncrCounter(CursorList *cl) {
   }
 }
 
+#define mask31(x) ((x) & 0x7fffffffUL) // mask to prevent overflow when adding 1 to the id
+#define rand_even48() (mask31(lrand48()) & ~(1UL))
+#define rand_odd48() (mask31(lrand48()) | (1UL))
```

Comment:
We need to protect from the case that `lrand48` returns the max long value (2^31-1) (actually also 2^31-2 since we set the first bit) and then we do +1 and wrap.
The 32nd bit is already cleared for these 2 values - so the mask has no effect?

Reply:
@oshadmi max long is 2^32-1.

Reply:
@oshadmi the long in is being implicit converted to Unsigned long by the compiler so there is no overflow.